rules/import_middleman.bzl

```diff
@@ -239,11 +249,17 @@ def _file_collector_rule_impl(ctx):
             dynamic_framework_dirs.append(f)
     objc_provider_fields["dynamic_framework_file"] = depset(dynamic_framework_file)
 
-    replaced_frameworks = replaced_dyanmic_framework.values() + replaced_static_framework.replaced.values()
+    all_replaced_frameworks = replaced_dyanmic_framework.values() + replaced_static_framework.replaced.values()
+
+    # This slightly varies in Bazel 6.x.x
+    if use_lts_5_rules_apple_api:
+        replaced_frameworks = all_replaced_frameworks
+    else:
+        replaced_frameworks = replaced_dyanmic_framework.values()
 
     compat_link_opt = ["-L__BAZEL_XCODE_DEVELOPER_DIR__/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator", "-Wl,-weak-lswiftCompatibility51"]
 
-    if len(replaced_frameworks):
+    if len(all_replaced_frameworks):
```

Comment:
Is it intentional that `linkopt` below is still using `replaced_frameworks` instead of `all_replaced_frameworks`?

```
["\"\"\"-F" + "/".join(f.path.split("/")[:-2]) + "\"\"\"" for f in replaced_frameworks] + compat_link_opt,
```

Reply:
That is the gist of it -  6.x.x `static_framework` was transitioned to `imported_library` and is in the `filelist` not here